<a href="https://colab.research.google.com/github/aviralsingh2907/PlayStore-App-Review-Analysis/blob/main/EDA_Play_Store_App_Review_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <b><u>Exploratory Data Analysis</u> - <u>Play Store App Review Analysis</u></b>

##### **Project Type**    - EDA
##### **Contribution**    - Individual
##### **Team Member**     - Aviral Singh

# <b><u>Project Summary</u>:</b>

# **<u>GitHub Link</u>:**

# <b><u>Problem Statement</u>:</b>

The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market.

Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.

Explore and analyze the data to discover key factors responsible for app engagement and success.

# <b><u>Dataset Import</u>:</b>

In [1]:
# Importing package:

import pandas as pd
import numpy as np
from google.colab import drive
import matplotlib.pyplot as plt
import matplotlib as mpl
import seaborn as sns


drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# Importing the datasets:

playstore_df_main = pd.read_csv('/content/drive/MyDrive/Portfolio Project/EDA - Play Store App Review/Play Store Data.csv')
user_review_df_main = pd.read_csv('/content/drive/MyDrive/Portfolio Project/EDA - Play Store App Review/User Reviews.csv')

In [3]:
# Creating copies:

playstore_df = playstore_df_main.copy()
user_review_df = user_review_df_main.copy()

In [4]:
playstore_df.head(5)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [5]:
user_review_df.head(5)

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   
3  10 Best Foods for You         Works great especially going grocery store   
4  10 Best Foods for You                                       Best idea us   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN  
3  Positive                0.40                0.875000  
4  Positive                1.00                0.300000

In [6]:
# Checking dataset information:

playstore_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [7]:
user_review_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


In [8]:
# Null values check:

playstore_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [9]:
user_review_df.isnull().sum()

App                           0
Translated_Review         26868
Sentiment                 26863
Sentiment_Polarity        26863
Sentiment_Subjectivity    26863
dtype: int64

# <b><u>Data Cleaning (Play Store Dataset)</u>:</b>

Cleaning raw data before processing & analysis:

* Determining which parts of the data are incomplete, incorrect, inaccurate or irrelevant.
* Replacing, changing, or deleting the soiled or coarse data.

Creating a function to obtain necessary information about various dataset attributes.

The function can be recalled & reused in the future.

In [12]:
# Defining the function:

def necessary_info():
    null = pd.DataFrame(index = playstore_df.columns)
    null['data_type'] = playstore_df.dtypes
    null['null_count'] = playstore_df.isnull().sum()
    null['unique_count'] = playstore_df.nunique()
    return null

In [13]:
# Calling the function:

necessary_info()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           1             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33

### <b><u>'App' column</u>:</b>

Checking for duplicate values in the 'App' column.

In [14]:
print("No. of Unique App names = ", len(playstore_df['App'].unique()))
print("Total no. of App names = ", playstore_df.shape[0])
print("Duplicate App names = ", len(playstore_df['App']) - len(playstore_df['App'].unique()))

No. of Unique App names =  9660
Total no. of App names =  10841
Duplicate App names =  1181


In [17]:
# Looking for occurence of duplicate values:

playstore_df[playstore_df['App'] == 'Coloring book moana']

App        Category  Rating Reviews Size  Installs  \
1     Coloring book moana  ART_AND_DESIGN     3.9     967  14M  500,000+   
2033  Coloring book moana          FAMILY     3.9     974  14M  500,000+   

      Type Price Content Rating                     Genres      Last Updated  \
1     Free     0       Everyone  Art & Design;Pretend Play  January 15, 2018   
2033  Free     0       Everyone  Art & Design;Pretend Play  January 15, 2018   

     Current Ver   Android Ver  
1          2.0.0  4.0.3 and up  
2033       2.0.0  4.0.3 and up

In [19]:
playstore_df.drop_duplicates(subset = 'App', keep = 'first', inplace = True)

### <b><u>'Type' column</u>:</b>

In [20]:
# Unique set of values in 'Type' column:

playstore_df.Type.unique()

array(['Free', 'Paid', nan, '0'], dtype=object)

0 & null values found.

Replacing them by 'Free'.

In [21]:
playstore_df['Type'].replace(to_replace=['0'], value=['Free'], inplace=True)
playstore_df['Type'].fillna('Free', inplace=True)

### <b><u>'Content Rating' column</u>:</b>

In [22]:
# Checking for null values:

playstore_df[playstore_df['Content Rating'].isnull()]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs  Type     Price Content Rating             Genres  \
10472  1,000+     Free  Free  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [23]:
# Data comparison with adjacent rows:

playstore_df.loc[10465:10477, :]

App         Category  Rating  \
10465              Portable Wi-Fi hotspot Free    COMMUNICATION     4.0   
10466                           Wi-Fi Networks     PRODUCTIVITY     4.2   
10467                                   FI CFL          FINANCE     3.7   
10468                         Tassa.fi Finland        LIFESTYLE     3.6   
10469              TownWiFi | Wi-Fi Everywhere    COMMUNICATION     3.9   
10470                               Jazz Wi-Fi    COMMUNICATION     3.4   
10471                         Xposed Wi-Fi-Pwd  PERSONALIZATION     3.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   
10474                             Sat-Fi Voice    COMMUNICATION     3.4   
10475                         Wi-Fi Visualizer            TOOLS     3.9   
10476                    Lennox iComfort Wi-Fi        LIFESTYLE     3.0   
10477              Sci-Fi Sounds and Ringtones  PERSONALIZATION     3.6   

      Reviews    Size  Installs  Type     Price Content Rating  \
10465    1711    2.1M  100,000+  Free         0       Everyone   
10466     115    1.4M   10,000+  Free         0       Everyone   
10467     112    3.9M   10,000+  Free         0       Everyone   
10468     346    7.5M   50,000+  Free         0       Everyone   
10469    2372     58M  500,000+  Free         0       Everyone   
10470      49    4.0M   10,000+  Free         0       Everyone   
10471    1042    404k  100,000+  Free         0       Everyone   
10472    3.0M  1,000+      Free  Free  Everyone            NaN   
10474      37     14M    1,000+  Free         0       Everyone   
10475     132    2.6M   50,000+  Free         0       Everyone   
10476     552    7.6M   50,000+  Free         0       Everyone   
10477     128     11M   10,000+  Free         0       Everyone   

                  Genres        Last Updated Current Ver   Android Ver  
10465      Communication    January 11, 2018  1.2.5.4-11    2.3 and up  
10466       Productivity     October 5, 2015         1.4    4.1 and up  
10467            Finance        July 5, 2018       1.1.1    5.0 and up  
10468          Lifestyle        May 22, 2018         5.5    4.0 and up  
10469      Communication      August 2, 2018       4.2.1    4.2 and up  
10470      Communication   February 10, 2017         0.1    2.3 and up  
10471    Personalization      August 5, 2014       3.0.0  4.0.3 and up  
10472  February 11, 2018              1.0.19  4.0 and up           NaN  
10474      Communication   November 21, 2014     2.2.1.5    2.2 and up  
10475              Tools        May 17, 2017       0.0.9    2.3 and up  
10476          Lifestyle      March 22, 2017      2.0.15  2.3.3 and up  
10477    Personalization  September 27, 2017         4.0    4.0 and up

<u>Observations</u>:

* Wrong data entry in row 10472.
* All of the column values found to be replaced by their previous columns' values due to a missing value in the category column.

In [24]:
# Dropping the rows containing null values in 'Content Rating' column:

playstore_df.dropna(subset=['Content Rating'], inplace=True)

### <b><u>'Rating' column</u>:</b>

* 'Rating' column has a total of 1463 missing values.
* Replacing the missing values with the Mode value of the column.

In [26]:
# Finding the mode value:

mode_rating = playstore_df['Rating'].mode()
print(f' The mode value of Rating column is: {mode_rating[0]}')

# Replacing the null values:

playstore_df['Rating'].fillna(value=mode_rating[0], inplace = True)
necessary_info()

 The mode value of Rating column is: 4.3


data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews           object           0          5330
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           118
Last Updated      object           0          1377
Current Ver       object           8          2817
Android Ver       object           2            33

### <b><u>Deleting Unnecessary Columns</u>:</b>

In [27]:
# Deleting unnecessary columns:

playstore_df.drop(['Current Ver','Last Updated', 'Android Ver'], axis=1, inplace=True)

In [28]:
necessary_info()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews           object           0          5330
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           118

* The null counts for all necessary columns are zero.
* There are no more missing values in the data frame.

# <b><u>Data Preparation (Play Store Dataset)</u>:</b>

* Verifying the data type for 'Reviews', 'Size', 'Installs' & 'Price' columns.
* They should ideally be 'int' or 'float'.
* Making appropriate changes if not the case.

### <b><u>'Reviews' column</u>:</b>

In [29]:
# Converting 'Reviews' data type to 'int':

playstore_df['Reviews'] = playstore_df.Reviews.astype(int)

### <b><u>'Size' column</u>:</b>

* String values like 'Varies with device', 'M' & 'k' present in 'Size' column.
* Converting KBs to MBs. Removing non-numeric values. Converting to 'int' or 'float'.

In [30]:
# Removing '+' :

playstore_df['Size'] = playstore_df.Size.apply(lambda x: x.strip('+'))

In [31]:
# Removing ',' :

playstore_df['Size'] = playstore_df.Size.apply(lambda x: x.replace(',', ''))

In [32]:
# Removing 'M' & 'k', and converting KBs into MBs:

playstore_df['Size'] = playstore_df['Size'].apply(lambda x: x.replace('M', '') if 'M' in str(x) else x )
playstore_df['Size'] = playstore_df['Size'].apply(lambda x: float(x.replace('k', ''))/1024 if 'k' in str(x) else x)

In [33]:
# Replacing 'Varies with device' by Nan:

playstore_df['Size'] = playstore_df.Size.replace('Varies with device', np.NaN)

* Null values need to be taken care of.
* Some apps' sizes will be too huge and others excessively small, i.e., too much variation in the apps' sizes.
* It's best to remove the rows containing Null value instead of replacing them with mean or mode.

In [34]:
# Removing rows containing Null (earlier 'Varies with device') value in 'Size' column:

playstore_df.dropna(subset=['Size'], inplace=True)

In [35]:
# Renaming the column:

playstore_df.rename(columns={'Size': 'Size(in MB)'}, inplace=True)

In [36]:
# Converting 'Size' datatype to numeric:

playstore_df['Size(in MB)'] = pd.to_numeric(playstore_df['Size(in MB)'])

### <b><u>'Installs' column</u>:</b>

In [37]:
# Checking for unique values:

playstore_df.Installs.unique()

array(['10,000+', '500,000+', '5,000,000+', '50,000,000+', '100,000+',
       '50,000+', '1,000,000+', '10,000,000+', '5,000+', '100,000,000+',
       '1,000+', '500,000,000+', '50+', '100+', '500+', '10+', '1+', '5+',
       '1,000,000,000+', '0+'], dtype=object)

* " , " needs to be replaced.
* " + " needs to be removed.
* Datatype needs to be changed to 'float'.

In [38]:
# Removing  '+' :

playstore_df['Installs'] = playstore_df.Installs.apply(lambda x: x.strip('+'))

# Replacing ',' :

playstore_df['Installs'] = playstore_df.Installs.apply(lambda x: x.replace(',', ''))

In [39]:
# Converting datatype to numeric:

playstore_df['Installs'] = pd.to_numeric(playstore_df['Installs'])

### <b><u>'Price' column</u>:</b>

In [ ]:
#checking for value count
play_store_df['Price'].value_counts()

In [ ]:
#Removing "$" sign
play_store_df['Price'] = play_store_df.Price.apply(lambda x: x.strip('$'))

In [ ]:
# converting to Numeric type
play_store_df['Price'] = pd.to_numeric(play_store_df['Price'])

Lets take a final look at our DataFrame

In [ ]:
#calling the function
complete_info()

In [ ]:
play_store_df.shape

There are 8434 rows and 10 columns left in the data frame after the dataset has been cleaned up by removing any unneeded rows and columns containing Null Values and garbage data.

## <b><u>Dataset summary based on</u>:</b>

* Total size occupied by each app category.
* Average rating for each app category.
* Total installs for each app category.
* Total reviews for each app category.

In [ ]:
categorical_summary = play_store_df.groupby('Category').agg({'Size(in MB)':'sum', 'Rating':'mean', 'Installs':'sum','Reviews':'sum'})
categorical_summary

# <b><u>Data Analysis & Visualization (Play Store Dataset)</u>:</b>

### <b><u>'Category' wise Analysis</u>:</b>

Well, let us try to find what are the top categories in the play store, which contains the highest number of apps?

In [ ]:
#unique categories
len(play_store_df['Category'].unique())

So we got 33 category on this dataset, let's see which one is the famous category

In [ ]:
# Determining top categories in data
x = play_store_df['Category'].value_counts().index
y = play_store_df['Category'].value_counts()
xaxis = []
yaxis = []
for i in range(len(y)):
    xaxis.append(x[i])
    yaxis.append(y[i])

In [ ]:
# Plotting graph/visuals for the same
plt.figure(figsize=(18,5))
plt.xlabel("Category", fontsize = 15)
plt.ylabel("Count", fontsize = 15)
plt.xticks(rotation=90)
category_graph = sns.barplot(x = xaxis, y = yaxis, palette= "rainbow")
category_graph.set_title("Categories of apps in google Playstore", fontsize = 25);

There are a total of 33 categories in the dataset, and based on the result from the previous step, we can infer that the majority of the apps in the Google Play store fall into the Family & Games category, while the least number of them fall into the Beauty & Comics category.

In [ ]:
#Finding the top 10 categories
Top10_categories=play_store_df['Category'].value_counts().reset_index().head(10)
Top10_categories.rename(columns={'index':'Category','Category':'Count'},inplace=True)
Top10_categories

In [ ]:
#Plotting Distribution of top 10 categories
plt.figure(figsize=(8,10))
plt.pie(Top10_categories['Count'],labels=Top10_categories['Category'],autopct='%.0f%%',explode=[0.02]*10)
plt.title('Top 10 categories distribution', fontsize= 20)
plt.show()

Among Top 10 Categories Family (31%), Games (15%) And Tools (13%) Contribute The Most

### <b><u>'Rating' wise Analysis</u>:</b>

In [ ]:
# ploting distribution graph for Rating
plt.figure(figsize=(14,5))
sns.distplot(play_store_df['Rating'],color = 'blue')
plt.grid()
plt.xlabel("Rating")
plt.ylabel("Frequency")
plt.title('Distribution Plot Of Rating')
plt.show()

In [ ]:
#calculating the average rating
print('The average rating in the playstore is',play_store_df['Rating'].mean())

This average rating infers that maximum number of apps in playstore have been rated above 4 which signifies majority of apps available in playstore are of high calibre.

In [ ]:
#App's With 5 Star Rating
five_star_rating_apps=play_store_df[play_store_df['Rating']==5]
five_star_rating_apps['App'].nunique()

In [ ]:
#Top 10 Categories Of 5 Star Rating App's
five_star_rating_apps['Category'].value_counts().reset_index().rename(columns={'index':'Category','Category':'Count'}).head(10)

### <b><u>'Content Rating' wise Analysis</u>:</b>

Let us see, which category of Apps from the ‘Content Rating’ column is found more on the play store.

In [ ]:
#Content rating value counts
value_c=play_store_df["Content Rating"].value_counts().reset_index()

In [ ]:
#barplot of content rating value counts
sns.barplot(x="Content Rating",y="index",data=value_c)
plt.title("Barplot of Content Rating ",fontsize=20)
plt.xlabel("No. of apps", fontsize= 15)
plt.ylabel("Content rating", fontsize= 15)

The Everyone category has the most apps, as can be seen from the plot above.

### <b><u>'Free' vs 'Paid' App Analysis</u>:</b>

In [ ]:
#plotting a graph between free and paid apps
plt.figure(figsize=(8,8))
labels = play_store_df['Type'].value_counts(sort = True).index
sizes = play_store_df['Type'].value_counts(sort = True)
colors = ["orange","blue"]
explode = (0.2,0)
plt.pie(sizes, explode=explode, labels=labels, colors=colors, autopct='%1.1f%%', shadow=True, startangle=0)
plt.title('Percentage of Free Vs Paid Apps in store',size = 20)
plt.show()

We can see from the graph above that 92% of the apps in the Google Play store are free, while 8% are paid.

### <b><u>'Installs' wise Analysis</u>:</b>

Let us check, which category App’s have the most number of installs?

In [ ]:
# defining x
x = play_store_df.groupby('Category')['Installs'].agg(np.sum)

In [ ]:
# plotting line graph to determine category highest installations
plt.figure(figsize=(18,8))
plt.plot(x ,  color='blue', marker='.')
plt.xticks(rotation=90)
plt.xlabel('Categories---->')
plt.ylabel('Installs---->')
plt.title('Category vs Installs')
plt.grid()
plt.show()

Graph clearly shows that apps in game category have been installed the highest followed by apps in family category and then apps in travel and local.

In [ ]:
# top 10 install app based on category
def top10incategory_installs(str):
    str = str.upper()
    top10 = play_store_df[play_store_df['Category'] == str]
    top10apps = top10.sort_values(by='Installs', ascending=False).head(10)
    # Top_Apps_in_art_and_design
    plt.figure(figsize=(15,12))
    plt.title('Top 10 Installed Apps',size = 20);
    plt.xlabel('App', fontsize= 15)
    plt.ylabel('Installs',fontsize= 15)
    graph = sns.barplot(x = top10apps.App, y = top10apps.Installs)
    graph.set_xticklabels(graph.get_xticklabels(), rotation= 45, horizontalalignment='right');

After we are done with defining the function, it’s time to check and see if everything is working fine. So let’s test it by passing Game category to the above-defined function.

In [ ]:
#calling the function
top10incategory_installs('Game')

Subway Surfers has the most instals in the Game category, as shown in the graph above. In the same way, we can obtain the top 10 installed apps by passing various category names to the function.

### <b><u>'Price' wise Analysis</u>:</b>

Let us Visualize, which are the top 10 expensive Apps in the play store?

In [ ]:
# We will again need to create a separate data frame.
top10PaidApps_df = play_store_df[play_store_df['Type'] == 'Paid'].sort_values(by='Price', ascending=False).head(11)

In [ ]:
#ploting the top 10 expensive App
plt.figure(figsize=(12,9));
plt.pie(top10PaidApps_df.Installs, explode=None, labels=top10PaidApps_df.App, autopct='%1.1f%%', startangle= -50);
plt.title('Top Expensive Apps Distribution',size = 20);
plt.legend(top10PaidApps_df.App,
           loc="top left",
           title="Apps",
           fontsize = "xx-small"
          );

From the above graph, we can interpret that the App 'I am rich Premium' is the most expensive app in the google play store followed by 'I am Rich'.

In [ ]:
#Discription of feature Price
print('The mean price of an App in playstore is',play_store_df['Price'].mean())
print('The maximum price of an App in playstore is',play_store_df['Price'].max())
play_store_df['Price'].describe()

### <b><u>'Size' wise Analysis</u>:</b>

In [ ]:
# Plotting a line graph to determine size distribution
plt.figure(figsize=(10,5))
plt.xlabel("Size(in MB)")
plt.ylabel("Frequency")
plt.grid()
size_distribution_graph = sns.kdeplot(play_store_df['Size(in MB)'], color="lightgreen", shade = True)
plt.title('Average size',size = 20);

In [ ]:
print('The median size of an App in playstore is',play_store_df['Size(in MB)'].median())
print('The maximum size of an App in playstore is',play_store_df['Size(in MB)'].max())

In [ ]:
# Plotting a boxplot graph to determine size distribution
plt.figure(figsize=(10,5))
plt.xlabel("Size(in MB)")
plt.ylabel("No of apps")
plt.grid()
size_distribution_graph = sns.boxplot(play_store_df['Size(in MB)'], color="lightgreen")
plt.title('Average size',size = 20);

As we can see from the box plot above,

* 25% of apps are under 5 MB
* 25% of apps are between 30 Mb to 63 Mb
* Majority i.e. 50% lies between 5 Mb to 30 Mb.
* Many outliers are all the way upto 100 Mb
* Median is 12 Mb

### <b><u>'Reviews' wise Analysis</u>:</b>

Apps which are highly reviewed

In [ ]:
Apps_and_reviews= play_store_df.groupby('App')[['Reviews']].mean().sort_values('Reviews', ascending=False).head(10).reset_index()
sns.barplot(y = Apps_and_reviews['App'], x = Apps_and_reviews['Reviews'])
plt.title('Apps with highest ratings')
plt.show()

In [ ]:
print("Number of Apps with more than 1M reviews",play_store_df[play_store_df['Reviews'] > 1000000].shape[0])
print("\nTop 20 apps with most reviews: \n",play_store_df[play_store_df['Reviews'] > 1000000].sort_values(by = 'Reviews', ascending = False).head(20)['App'])

### <b><u>'Genres' wise Analysis</u>:</b>

In [ ]:
#top 10 Genres Value Counts
genres_count=play_store_df['Genres'].value_counts().reset_index()
genres_count.rename(columns={'index':'Genres','Genres':'count'},inplace=True)
top_10_genres=genres_count.head(10)

In [ ]:
#Pie plot of Top 15 genres Count
plt.rcParams['figure.figsize'] = (20, 10)
plt.pie(top_10_genres['count'],labels=top_10_genres['Genres'],autopct='%.0f%%')
plt.title('Pie plot of Top 10 genres Count')
plt.grid()
plt.show()

Tools is the most used genre in apps.

## <b><u>Correlation Heatmap</u>:</b>

In [ ]:
#Correlation
play_store_df.corr()

In [ ]:
#Correlation Heatmap
plt.figure(figsize = (10,6))
sns.heatmap(play_store_df.corr(), annot= True)
plt.title("Correlation Heatmap",fontsize=20)

In [ ]:
#Regression Plot Of Installs And Reviews
plt.figure(figsize = (10,6))
sns.regplot(x="Installs",y="Reviews",data=play_store_df)
plt.title("Regression Plot of Installs And Reviews",fontsize=25)
plt.grid()

# <b><u>User Reviews Dataset</u>:</b>

In [ ]:
#First Look Of User Reviews Data
user_review.head()

In [ ]:
user_review.info()

### <b><u>Handling missing values</u>:</b>

In [ ]:
#data of Translated null values
user_review[user_review['Translated_Review'].isnull()].head()

In [ ]:
#Dropping Nulls of Translated Review as all other feature values are also null
user_review.dropna(subset=['Translated_Review'],inplace=True)

In [ ]:
#Checking data after removing nulls
user_review.info()

# <b><u>Combining 'Play Store Dataset' & 'User Review Dataset'</u>:</b>

In [ ]:
#unique apps is user_reviews_data
user_review['App'].nunique()

In [ ]:
#unique apps is play_store_data
play_store_df['App'].nunique()

In [ ]:
#Merging both data
combined_data=pd.merge(play_store_df,user_review, on='App')

In [ ]:
#Unique Apps in combined data
combined_data['App'].nunique()

In [ ]:
#About combined data
combined_data.isnull().sum()

# <b><u>Analysis Of Combined Data</u>:</b>

### <b><u>'Sentiment' wise Analysis</u>:</b>

In [ ]:
#Sentiment count
sentiment_count= combined_data.Sentiment.value_counts().reset_index().rename(columns={'index':'Sentiment','Sentiment':'count'})
sentiment_count

In [ ]:
#Pieplot of Sentiment Count
plt.rcParams['figure.figsize'] = (15, 7)
plt.pie(sentiment_count['count'],labels=sentiment_count['Sentiment'],autopct='%.0f%%',explode=(0,0.05,0.05))
plt.title('Pie plot of Sentiment Count',size=15)
plt.show()

This chart interprets that most of the reviews are positive.

In [ ]:
#Categoriwise Sentiment Count
categoriwise_sentiment_count=combined_data.groupby(['Category','Sentiment'])['Sentiment'].count().reset_index(name='count')
categoriwise_sentiment_count.head()

In [ ]:
#Barplot of categoriwise_sentiment_count
sns.barplot(x="Category", y="count", hue="Sentiment", data=categoriwise_sentiment_count)
plt.xticks(rotation=90, horizontalalignment="center")
plt.title("Category-wise sentiment count",fontsize=20)
plt.xlabel("Category", fontsize= 15)
plt.ylabel("Sentiments", fontsize= 15)

This graph represents sentimnet analysis of the audience based on the categories, it can be observed that all the categories have more positive reviews than negatives.

# <b><u>Conclusion</u>:</b>

Joining the dots from given datasets, we came across a lot of information. After analysis and visualization, information can be classified into beneficial for customers and beneficial for developers. We discovered the top app categories, the most popular app categories, the proportion of free and paid apps, and the average app size. This information enables customers and users to make informed download decisions.

Game category has the highest engagement

• Thus, if fast growth is expected then introducing good quality game app of suitable size can work as a charm.

Opportunities

• Various apps category like Medical, food & drinks, Health & fitness and business & finances have positive sentimental advantage from users but the population of apps is less in those categories creating opportunities for new players